web-bluetooth/scan.js

```diff
@@ -0,0 +1,34 @@
+async function onButtonClick() {
+  let filters = [];
+
+  let filterName = document.querySelector('#name').value;
+  if (filterName) {
+    filters.push({name: filterName});
+  }
+
+  let filterNamePrefix = document.querySelector('#namePrefix').value;
+  if (filterNamePrefix) {
+    filters.push({namePrefix: filterNamePrefix});
+  }
+
+  let options = {};
+  if (document.querySelector('#allAdvertisements').checked) {
+    options.acceptAllAdvertisements = true;
+  } else {
+    options.filters = filters;
+  }
+
+  try {
+    log('Requesting Bluetooth Scan...');
+    log('with ' + JSON.stringify(options));
+    await navigator.bluetooth.requestLEScan(options);
```

Comment:
I think basic demo should at least allow users to stop scanning.
How about something like below?
```js
var scan;

async function onStartButtonClick() {
  /* ... */
  scan = await navigator.bluetooth.requestLEScan(options);
  /* ... */
}

async function onStopButtonClick() {
  if (!scan || !scan.active) {
    return;
  }
  scan.stop();
}
`

Reply:
I'd like to keep the code as simple as possible, I've used a simple timer to stop the scan after 10 seconds.